timesketch/lib/analyzers/win_crash.py

```diff
@@ -0,0 +1,195 @@
+            match = self.FILENAME_REGEX.search(text)
+            if match:
+                # The regex can match on full file paths and filenames,
+                # so only return the filename.
+                return min([m for m in match.groups() if m])
+        return None
+
+    def mark_as_crash(self, event, filename):
+        """Mark entries with crash artefacts.
+
+        Args:
+            event: Elasticsearch event
+            filename: Application that crashed
+        """
+        if filename:
+            event.add_attributes({'crash_app': filename})
+        event.add_tags(['win_crash'])
+
+    def run(self):
+        """Entry point for the analyzer.
+
+        Returns:
+            String with summary of the analyzer result
+        """
+        query = self.formulate_query(self.QUERY_ELEMENTS)
+
+        return_fields = ['data_type', 'message', 'filename']
+
+        # Generator of events based on your query.
+        events = self.event_stream(
+            query_string=query, return_fields=return_fields)
+
+        # Container for filenames of crashed applications.
+        filenames = list()
+
+        for event in events:
+            data_type = event.source.get('data_type')
+            event_text = None
+
+            # Search event log entries for filenames of crashed applications.
+            if data_type == 'windows:evtx:record':
+                event_text = event.source.get('message')
+
+            # Search file system entries for filenames of crashed applications.
+            elif data_type == 'fs:stat':
+                event_text = event.source.get('filename')
+
+            # Tag entries that show the crash reporting has been disabled.
+            elif data_type == 'windows:registry:key_value':
+                event.add_comment(
+                    'WARNING: The crash reporting was disabled. '
+                    'It could be indicative of attacker activity. '
+                    'For details refer to page 16 from '
+                    'https://assets.documentcloud.org/documents/3461560/'
+                    'Google-Aquarium-Clean.pdf.')
+                event.commit()
+
+            # If found the filename, tag the entry as crash-related
+            filename = self.extract_filename(event_text)
+            if filename:
+                self.mark_as_crash(event, filename)
+                filenames.append(filename)
+                event.commit()
+
+        # Dedup the filenames of crashed apps
+        filenames = set(filenames)
+
+        # Create a saved view with our query.
+        if filenames:
+            self.sketch.add_view(
+                'Windows Crash activity', 'win_crash', query_string=query)
+
+        return 'Windows Crash analyzer completed, ' + \
+            '{0:d} crashed application{1:s} identified: {2:s}'.format(
+                len(filenames),
+                's' if len(filenames) > 1 else '',
+                ', '.join(filenames))
```

Comment:
do we want all the filenames on this return string? what if there are 200 filenames returned, this string would make the logs completely unreadable, and it does not provide any value really, remember this is displayed in the ui andn is in the logs on the server, no need to contain specifics, more just the number of applications, etc

Reply:
I see your point. From testing a few Plaso files, I had a very small number of application names that were returned. Once the analyzer finished running, it was actually quite helpful for me to see the filenames that crashed in the web UI (iirc, it was called "history") before diving into the analysis. If you don't mind, I'd leave it as it is.

Reply:
ok, but we'll remove this if it becomes an issue